assets/js/src/integrations/classic.js

```diff
@@ -0,0 +1,132 @@
+import { tracker } from '../tracker';
+import { getProductFromID } from '../utils';
+import {
+	events,
+	cart,
+	products,
+	product,
+	addedToCart,
+	order,
+} from '../config.js';
+
+/**
+ * The Google Analytics integration for classic WooCommerce pages
+ * triggers events using three different methods.
+ *
+ * 1. Automatically handle events listed in the global `wcgaiData.events` object.
+ * 2. Listen for custom events from WooCommerce core.
+ * 3. Listen for various actions (i.e clicks) on specific elements.
+ */
+
+export const trackClassicIntegration = () => {
+	const eventData = {
+		storeCart: cart,
+		products,
+		product,
+		order,
+	};
+
+	Object.values( events ?? {} ).forEach( ( eventName ) => {
+		if ( eventName === 'add_to_cart' ) {
+			tracker.eventHandler( eventName )( { product: addedToCart } );
+		} else {
+			tracker.eventHandler( eventName )( eventData );
+		}
+	} );
+
+	/**
+	 * Track the custom add to cart event dispatched by WooCommerce Core
+	 *
+	 * @param {Event} e - The event object
+	 * @param {Object} fragments - An object containing fragments of the updated cart.
+	 * @param {string} cartHash - A string representing the hash of the cart after the update.
+	 * @param {HTMLElement[]} button - An array of HTML elements representing the add to cart button.
+	 */
+	document.body.onadded_to_cart = ( e, fragments, cartHash, button ) => {
+		tracker.eventHandler( 'add_to_cart' )( {
+			product: getProductFromID(
+				parseInt( button[ 0 ].dataset.product_id )
+			),
+		} );
+	};
+
+	/**
+	 * Attaches click event listeners to all remove from cart links
+	 */
+	const removeFromCartListener = () => {
+		document
+			.querySelectorAll(
+				'.woocommerce-cart-form .woocommerce-cart-form__cart-item .remove[data-product_id]'
+			)
+			.forEach( ( item ) =>
+				item.addEventListener( 'click', removeFromCartHandler )
+			);
+	};
+
+	/**
+	 * Handle remove from cart events
+	 *
+	 * @param {HTMLElement|Object} element - The HTML element clicked on to trigger this event
+	 */
+	function removeFromCartHandler( element ) {
+		tracker.eventHandler( 'remove_from_cart' )( {
+			product: getProductFromID(
+				parseInt( element.target.dataset.product_id )
+			),
+		} );
+	}
+
+	// Attach event listeners on initial page load and when the cart div is updated
+	removeFromCartListener();
+	document.body.onupdated_wc_div = () => removeFromCartListener();
+
+	// Trigger the handler when an item is removed from the mini-cart and WooCommerce dispatches the `removed_from_cart` event.
+	document.body.onremoved_from_cart = (
+		event,
+		fragments,
+		/* eslint-disable-next-line camelcase */
+		cart_hash,
+		button
+	) => removeFromCartHandler( { target: button[ 0 ] } );
```

Comment:
As we are not using `on` or `addEventListener`, but hard-setting the property, we could be more gently towards other plugins, trying to hook into those events:

```suggestion
	const oldOnupdatedWcDiv = document.body.onupdated_wc_div;
	document.body.onupdated_wc_div = ( ...args ) => {
		if ( typeof oldOnupdatedWcDiv === 'function' ) {
			oldOnupdatedWcDiv( ...args );
		}
		removeFromCartListener();
	};

	// Trigger the handler when an item is removed from the mini-cart and WooCommerce dispatches the `removed_from_cart` event.
	const oldOnRemovedFromCart = document.body.onremoved_from_cart;
	document.body.onremoved_from_cart = ( ...args ) => {
		if ( typeof oldOnRemovedFromCart === 'function' ) {
			oldOnRemovedFromCart( ...args );
		}
		removeFromCartHandler( { target: args[ 3 ][ 0 ] } );
	};
```